tests/eas/load_tracking.py

```diff
@@ -537,6 +540,406 @@ def test_load_avg_behaviour(self, experiment, tasks):
 
+class _CPUMigrationBase(LisaTest):
+    """Base class for shared functionality of tasks migration tests"""
+    test_conf = {
+        'tools'    : [ 'rt-app' ],
+        'ftrace' : {
+            'events' : [
+                'sched_switch',
+                'sched_load_cfs_rq',
+            ],
+        },
+        'modules': ['cpufreq', 'cgroups'],
+    }
+
+    # Allowed error margin
+    allowed_util_margin = UTIL_SCALE * 0.02
+
+    @classmethod
+    def setUpClass(cls, *args, **kwargs):
+        super(_CPUMigrationBase, cls).runExperiments(*args, **kwargs)
+
+    @classmethod
+    def generate_json(cls, test_env, tasks, workload_name):
+        """
+        Generate a json file which describes the different tasks.
+        :param test_env: Environment of the test
+        :type test_env: TestEnv
+
+        :param tasks: lists of task
+        :type tasks:[RTATask]
+
+        :param workload_name: name of the workload
+        :type workload_name: str
+        :return the json file name
+        """
+        rtapp = RTA(test_env.target, workload_name,
+            calibration=test_env.calibration())
+        rtapp.conf(kind='profile',
+            params={'task{}'.format(i): task.get() for i, task in enumerate(tasks)},
+            run_dir=test_env.target.working_directory)
+        return rtapp.json
+
+    @classmethod
+    def get_wload(cls, json_file):
+        """
+        Get a specification for a rt-app workload from a json file
+        :param json_file: path to the file that contains the tasks description
+        :type json_file: str
+
+        Example of expected task description:
+        "task0": {
+            "loop": 1,
+            "phases": {
+                "p1": {
+                    "cpus": [4],
+                    "loop": 62,
+                    "run": 4000,
+                    "timer": {
+                        "period": 16000,
+                        "ref": "task0"
+                    }
+                }
+            }
+        }
+        """
+
+        return {
+            'migration': {
+                'type' : 'rt-app',
+                'conf' : {
+                    'class' : 'custom',
+                    'json' : json_file,
+                    'prefix' : 'mig_test',
+                },
+            }
+        }
+
+    def _get_phases_names(self, tasks):
+        """
+        Return the phases names of a workload
+        :param task: Tasks description
+        :type task: dict
+        :return a list of the names of the phases
+        """
+        task_name = tasks.keys()[0]
+        return tasks[task_name]['phases'].keys()
+
+    def _get_duration_s(self, tasks, phase):
+        """
+        Compute the duration of a given phase
+        :param task: Tasks description
+        :type task: dict
+
+        :param phase: phase for which the duration is computed
+        :type phase: str
+        :return the duration in second of the phase
```

Comment:
Space before return?... same for following documentation

Reply:
Done